File: Top 100 Codes/02-Working with Numbers/08-Decimal to Octal Conversion/DecimaltoOctalConversion.py
# Function to convert a decimal number to its octal representation
def convert_to_octal(decimal):
    """
    Converts a decimal number to octal.
    
    Args:
        decimal (int): The decimal number to convert.
    
    Returns:
        int: The octal representation of the number.
    """
    octal = 0  # Variable to store the octal number
    position_multiplier = 1  # Multiplier for positional value in octal

    # Loop until the decimal number becomes 0
    while decimal > 0:
        # Get the remainder when dividing by 8
        remainder = decimal % 8

        # Update the octal number
        octal += remainder * position_multiplier

        # Move to the next positional multiplier (units -> tens -> hundreds, etc.)
        position_multiplier *= 10

        # Update the decimal number by dividing by 8
        decimal //= 8

    return octal


# Prompt the user for input
decimal_num = int(input("Enter a decimal number: "))

# Convert the decimal number to octal and display the result
octal_num = convert_to_octal(decimal_num)
print(f"The octal representation of {decimal_num} is: {octal_num}")

File: Top 100 Codes/02-Working with Numbers/08-Decimal to Octal Conversion/DecimaltoOctalConversion.cpp
#include <iostream>
using namespace std;

// Function to convert a decimal number to octal
void convertOctal(int num)
{
    // Variable to store the octal representation
    int octal = 0;
    int remainder, position = 1;

    // Loop until the decimal number becomes 0
    while (num != 0)
    {
        // Get the remainder when dividing by 8
        remainder = num % 8;

        // Divide the number by 8 to process the next digit
        num /= 8;

        // Add the remainder to the octal number at the current position
        octal += remainder * position;

        // Move to the next positional multiplier (units -> tens -> hundreds, etc.)
        position *= 10;
    }

    // Output the octal representation
    cout << "Octal representation: " << octal << endl;
}

int main()
{
    int decimal_num;

    // Prompt the user for input
    cout << "Enter a decimal number: ";
    cin >> decimal_num;

    // Function call to convert the decimal number to octal
    convertOctal(decimal_num);

    // Return 0 to indicate successful execution
    return 0;
}
